src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs

```diff
@@ -20,13 +19,10 @@ namespace OpenTelemetry.Instrumentation.AWSLambda;
 /// </summary>
 public static class AWSLambdaWrapper
 {
-    private static readonly AssemblyName AssemblyName = typeof(AWSLambdaWrapper).Assembly.GetName();
+    internal const string ActivitySourceName = "OpenTelemetry.Instrumentation.AWSLambda";
 
-    [SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1202:ElementsMustBeOrderedByAccess", Justification = "Initialization order.")]
-    internal static readonly string ActivitySourceName = AssemblyName.Name;
-
-    private static readonly Version Version = AssemblyName.Version;
-    private static readonly ActivitySource AWSLambdaActivitySource = new(ActivitySourceName, Version.ToString());
+    private static readonly string Version = typeof(AWSLambdaWrapper).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()!.Version;
+    private static readonly ActivitySource AWSLambdaActivitySource = new(ActivitySourceName, Version);
```

Comment:
I'll double-check in an application of mine that uses native AoT this won't get trimmed away now, but I think this is the best way to resolve this question.

`Assembly.GetName()` looks like mainly a vehicle to get the name and version, but that won't work for native AoT.

If `[AssemblyFileInfo]` is not trimmed, we can use that for the version, and changing the name of the assembly is going to be a breaking change (and probably won't change again?) so should be safe enough to hard-code. It also ensures that this cod never gets a `null`, which would break stuff if it did:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/8a89a69445a1f89917a6a159e895b9267c7cc925/src/OpenTelemetry.Instrumentation.AWSLambda/TracerProviderBuilderExtensions.cs#L42

Reply:
Yep, getting the assembly attribute appears to work.

Reply:
I wonder if the call `Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()!.Version `still works and what it returns for native AoT. Did you test it just with a simple AoT application ? Hopefully this list is complete: https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli#api-incompatibility

Reply:
I have a native AoT published ASP.NET Core application which returns metadata about its execution context, and this works correctly:

```csharp
        endpoints.MapGet("/configuration/version", () =>
        {
            return new JsonObject()
            {
                ["frameworkDescription"] = JsonValue.Create(RuntimeInformation.FrameworkDescription),
                ["operatingSystem"] = new JsonObject()
                {
                    ["description"] = JsonValue.Create(RuntimeInformation.OSDescription),
                    ["architecture"] = JsonValue.Create(RuntimeInformation.OSArchitecture.ToString()),
                    ["version"] = JsonValue.Create(Environment.OSVersion.VersionString),
                    ["is64Bit"] = JsonValue.Create(Environment.Is64BitOperatingSystem),
                },
                ["process"] = new JsonObject()
                {
                    ["architecture"] = JsonValue.Create(RuntimeInformation.ProcessArchitecture.ToString()),
                    ["is64BitProcess"] = JsonValue.Create(Environment.Is64BitProcess),
                    ["isNativeAoT"] = JsonValue.Create(!RuntimeFeature.IsDynamicCodeSupported),
                    ["isPrivilegedProcess"] = JsonValue.Create(Environment.IsPrivilegedProcess),
                },
                ["dotnetVersions"] = new JsonObject()
                {
                    ["runtime"] = JsonValue.Create(GetVersion<object>()),
                    ["aspNetCore"] = JsonValue.Create(GetVersion<HttpContext>()),
                },
            };
        }).ExcludeFromDescription();

        static string GetVersion<T>()
            => typeof(T).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()!.InformationalVersion;
```

I adapted the code to check `[AssemblyFileVersion]` earlier - can test it again if necessary with `AssemblyInformationalVersionAttribute` specifically from this assembly.

Reply:
The response:

```json
{
  "frameworkDescription": ".NET 8.0.1",
  "operatingSystem": {
    "description": "Ubuntu 22.04.3 LTS",
    "architecture": "X64",
    "version": "Unix 5.10.184.175",
    "is64Bit": true
  },
  "process": {
    "architecture": "X64",
    "is64BitProcess": true,
    "isNativeAoT": true,
    "isPrivilegedProcess": false
  },
  "dotnetVersions": {
    "runtime": "8.0.1+bf5e279d9239bfef5bb1b8d6212f1b971c434606",
    "aspNetCore": "8.0.1+8e941eb42f819adb116b881195158b3887a70a1c"
  }
}
```

Reply:
If it works for `AssemblyInformationalVersionAttribute` then should be fine for `AssemblyFileVersion`. No need for double check.